Automattic/wp-calypso
# Reader: refactor infinite-stream component

This PR seeks to create a new `InfiniteLoader` esque component implemented with react-virtualized.  That way we can reuse it throughout reader code.

This PR creates the new component and replaces `SitesWindowScroller` with it.
It will also be used in https://github.com/Automattic/wp-calypso/pull/13198

Things I can see us doing with this component in the future:
- ~~adding metrics~~ done
- adding arrow key support through https://github.com/bvaughn/react-virtualized/blob/master/docs/ArrowKeyStepper.md
- adding the ability to render posts as well so that it could be used in tandem with the reduxified streams.  it should ideally be as simple as adding a new row-renderer

TODO
- [x] fix loading bug
- [x] create the composable version of rowRenderer

To test:
1. load this branch 
2. go to /following/manage and make sure all of the lists there work as expected. that includes searching followed, scrolling through followed, searching feeds, and clicking show more to see more feeds.
3. go to /following/search and make sure that site results still render properly (should also be more performant)
3. railcars should also be fired for following-manage
